samza-core/src/main/java/org/apache/samza/util/DiagnosticsUtil.java

```diff
@@ -144,7 +145,8 @@ public MetadataFileContents(String version, String metricsSnapshot) {
       }
       SystemFactory systemFactory = ReflectionUtil.getObj(diagnosticsSystemFactoryName.get(), SystemFactory.class);
       SystemProducer systemProducer =
-          systemFactory.getProducer(diagnosticsSystemStream.getSystem(), config, new MetricsRegistryMap());
+          systemFactory.getProducer(diagnosticsSystemStream.getSystem(), config, new MetricsRegistryMap(),
+              MethodHandles.lookup().lookupClass().getSimpleName());
```

Comment:
i think it is ok to hardcode the prefix to "DiagnosticsUtil" too.

Reply:
Why not use Class.getSimpleName() here too?

Reply:
I think it is a static method. And we CAN use simpleName(), but we don't have too. As long as it is unique for the caller.

Reply:
Got it, thanks. We can probably use `DiagnosticsUtil.class.getSimpleName()` so that any refactoring updates it.

Reply:
Got it. Will do.